iroh-rpc/src/server.rs

```diff
@@ -0,0 +1,568 @@
+                                        )
+                                        .expect("Connection to peer to still be open.");
+                                    return;
+                                }
+                            };
+                        }
+                        CoreRequestEvent::Packet(packet) => {
+                            let stream_id = packet.id;
+                            let index = packet.index;
+                            debug!(target: "inbound streaming", "Received Packet {} for stream {}", index, stream_id);
+                            let sender = match self.active_streams.get_mut(&stream_id) {
+                                Some(s) => s,
+                                None => {
+                                    error!(target: "inbound streaming", "Packet {} could not be delievered because stream {} no longer in active stream list", index, stream_id);
+                                    self.swarm
+                                        .behaviour_mut()
+                                        .core
+                                        .send_response(
+                                            channel,
+                                            CoreResponse(CoreResponseEvent::RPCError(
+                                                RPCError::StreamClosed,
+                                            )),
+                                        )
+                                        .expect("Connection to peer to still be open.");
+                                    return;
+                                }
+                            };
+
+                            if let Err(e) = sender.try_send(StreamType::Packet(packet)) {
+                                // TODO: recover from error
+                                // send error back instead of panicing
+                                error!(target: "inbound streaming", "Error sending packet from wire to stream {}: {}", stream_id, e);
+                                panic!("Error sending packet off the wire to stream {}", stream_id);
+                            }
+                            // only ack for now, send back error if we cannot send to stream
+                            debug!(target: "inbound streaming", "Acknowledging Packet {} from stream {}", index, stream_id);
+                            self.swarm
+                                .behaviour_mut()
+                                .core
+                                .send_response(channel, CoreResponse(CoreResponseEvent::Ack))
+                                .expect("Connection to peer to still be open.");
+                        }
+                    },
+                    RequestResponseMessage::Response {
+                        request_id,
+                        response,
+                    } => match response.0 {
+                        CoreResponseEvent::Payload(payload) => {
+                            let _ = self
```

Comment:
this logic is replicated over all `match`  statements, you could just have that at the top, and then do the sending in the `match`